Admin/AbstractAdminDefinition.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Knp\Menu\ItemInterface as MenuItemInterface;
+use Sonata\AdminBundle\Datagrid\DatagridMapper;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\AdminBundle\Show\ShowMapper;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractAdminDefinition implements AdminDefinitionInterface, AdminHookInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getExportFormats()
+    {
+        return array(
+            'json',
+            'xml',
+            'csv',
+            'xls',
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preValidate($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preUpdate($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postUpdate($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function prePersist($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postPersist($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preRemove($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function postRemove($object)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureBatchActions(&$actions)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function preBatchAction($actionName, ProxyQueryInterface $query, array &$idx, $allElements)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureFormFields(FormMapper $formMapper)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureListFields(ListMapper $listMapper)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureDatagridFilters(DatagridMapper $datagridMapper)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureShowFields(ShowMapper $showMapper)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureRoutes(RouteCollection $routeCollection)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureTabMenu(MenuItemInterface $menu, $action, AdminInterface $childAdmin = null)
+    {
+    }
```

Comment:
What I miss here:
- method for specifying form options - e.g. validation_groups (currently it can be done with overloading `getFormBuilder()`), Maybe this can be included in the `FormMapper` so this method can be omitted.

- method for presetting values to new instance of $object - currently `getNewInstance()`. We can create  `postInitialize($object)` that will be called just after the entity is created,

- method for getting request query attributes or better `Request` itself - some logic in methods can be based on request attributes (important if route pattern is overloaded), Can be omitted. If user need this, he can add it in a setter. But this can be common use case and requested feature.

- method for customization query used in list - currently `createQuery()` - import if you need to not display items for some user roles, This can be separated outside from admin to model layer. But we have to think about it if the model layer will be refactored.

- method for getting currently edited object - currently `getSubject()`. It can be omitted, if we move this as a new parameter in the `configureFormFields` and `configureShowFields`, e.g.:

```
public function configureFormFields(FormMapper $formMapper, $object)
public function configureShowFields(ShowMapper $showMapper, $object)
```